python/pyspark/sql/session.py

```diff
@@ -470,6 +470,14 @@ def getOrCreate(self) -> "SparkSession":
                         ):
                             url = opts.get("spark.remote", os.environ.get("SPARK_REMOTE"))
 
+                            if url is None:
+                                raise RuntimeError(
```

Comment:
side note: we might want to move all these errors in this file to use error classes cc @itholic

Reply:
Hm, `SPARK_CONNECT_MODE_ENABLED` is an internal environment variable so I would like to avoid mentioning directly in the error message.

Reply:
I think you can just simply say that should set `SPARK_REMOTE` or `spark.remote` option if you intend to enable Spark Connect.